src/ui/components/inputs/DateInput.tsx

```diff
@@ -78,11 +79,19 @@ export const DateInput: FunctionComponent<DateInputProps> = (props) => {
       yearInputRef?.current?.focus()
     }
   }, [month])
-  useEffect(() => {
-    if (yearValidator.hasRightLength(year)) {
-      yearInputRef?.current?.blur()
+
+  /** focus previous input after deleting all the content: only for month and year */
+  function handleBackSpace(
+    e: NativeSyntheticEvent<TextInputKeyPressEventData>,
+    identifier: DatePartType
+  ) {
+    if (e.nativeEvent.key !== 'Backspace') {
+      return
     }
-  }, [year])
+
+    identifier === DatePartType.MONTH && month.length <= 1 && dayInputRef?.current?.focus()
+    identifier === DatePartType.YEAR && year.length <= 1 && monthInputRef?.current?.focus()
```

Comment:
Utilise un `if` pour une execution conditionelle, car les short-circuits sont pas faciles à lire :
```suggestion
    if (identifier === DatePartType.MONTH && month.length <= 1) {
      dayInputRef?.current?.focus()
    }
    if (identifier === DatePartType.YEAR && year.length <= 1) {
      monthInputRef?.current?.focus()
    }
```